engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -2116,9 +2122,24 @@ namespace dmGameSystem
 
     dmGameObject::PropertyResult CompGuiSetProperty(const dmGameObject::ComponentSetPropertyParams& params) {
         GuiComponent* gui_component = (GuiComponent*)*params.m_UserData;
-        if (params.m_PropertyId == PROP_MATERIAL) {
+        dmhash_t set_property = params.m_PropertyId;
+        if (set_property == PROP_MATERIAL) {
             return SetResourceProperty(dmGameObject::GetFactory(params.m_Instance), params.m_Value, MATERIAL_EXT_HASH, (void**)&gui_component->m_Material);
         }
+        else if (set_property == PROP_FONT) {
+            dmRender::HFontMap font;
+            memset(&font, 0, sizeof(dmRender::HFontMap));
+            dmGameObject::PropertyResult res = SetResourceProperty(dmGameObject::GetFactory(params.m_Instance), params.m_Value, FONT_EXT_HASH, (void**)&font);
+            dmGui::HScene scene = gui_component->m_Scene;
+            dmGui::Result r = dmGui::AddFont(scene, params.m_Value.m_Hash, (void*) font, params.m_Value.m_Hash);
```

Comment:
We want to operate these fonts as a regular font, so we add it in GUI fonts using resource path hash as name, because we doesn't have any other name. I think developer knows himself what font in what property he has.

So it may be used like this:

`font_prop.script`
```
go.property("font", resource.font())
function init(self)
	go.set("#gui", "font", self.font)
	msg.post("#font", "change", {font = self.font})
end
```

`font_prop.gui_script`
```
function on_message(self, message_id, message, sender)
	if message_id == hash("change") then
		gui.set_font(gui.get_node("text"), message.font)
	end
end
```

Reply:
It's not really a "set" here, but an "add"

I think this api leaves too much ambiguity. E.g.:
> go.set("#gui", "font", self.font1)
> go.set("#gui", "font", self.font2)

Which font is now "set" ?

I'm not sure what's the best way to do it is here, but as a worst case, we should be able to use a new message:
> msg.post(gui_url, "add_resource", { alias = alias, resource = hash, type = ".fontc" }

(probably possible to detect extension programattically fromt he resource system)

Or possibly separate messages: "add_font"/"add_texture"

There's also the issue of "removing" references (if we need it)
E.g. clearing references.